orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/RunTaskHandler.kt

```diff
@@ -128,36 +131,34 @@ class RunTaskHandler(
   }
 
   private fun trackResult(stage: Stage, taskModel: com.netflix.spinnaker.orca.pipeline.model.Task, status: ExecutionStatus) {
-    val id = registry.createId("task.invocations")
-      .withTag("status", status.toString())
-      .withTag("executionType", stage.execution.type.name.capitalize())
-      .withTag("taskType", taskModel.implementingClass)
-      .withTag("isComplete", status.isComplete.toString())
+    val commonTags = arrayListOf(
+      BasicTag("status", status.toString()),
+      BasicTag("executionType", stage.execution.type.name.capitalize()),
+      BasicTag("isComplete", status.isComplete.toString()),
+      BasicTag("cloudProvider", stage.context["cloudProvider"].toString()?: "n/a"))
+
+    val taskInvocationsId = registry.createId("task.invocations")
+      .withTags(commonTags)
       .withTag("application", stage.execution.application)
-      .let { id ->
-        stage.context["cloudProvider"]?.let {
-          id.withTag("cloudProvider", it.toString())
-        } ?: id
-      }
-    registry.counter(id).increment()
 
-    val distributionId = registry.createId("task.invocations.duration").withTags(id.tags())
-    BucketCounter
-      .get(registry, distributionId, { v -> bucketDuration(v) })
-      .record(System.currentTimeMillis() - (taskModel.startTime ?: 0))
-  }
+    val taskInvocationsWithTypeId = registry.createId("task.invocations.withType")
+      .withTags(commonTags)
+      .withTag("taskType", taskModel.implementingClass)
+      .withTag("account", stage.context["account"].toString()?: "n/a")
+      .withTag("region", stage.context["region"].toString()?: "n/a")
+
+    registry.counter(taskInvocationsId).increment()
+    registry.counter(taskInvocationsWithTypeId).increment()
+
+    val elapsedMillis = System.currentTimeMillis() - (taskModel.startTime ?: 0)
 
-  fun bucketDuration(duration: Long): String {
-    return if (duration > TimeUnit.MINUTES.toMillis(60)) {
-      "gt60m"
-    } else if (duration > TimeUnit.MINUTES.toMillis(30)) {
-      "gt30m"
-    } else if (duration > TimeUnit.MINUTES.toMillis(15)) {
-      "gt15m"
-    } else if (duration > TimeUnit.MINUTES.toMillis(5)) {
-      "gt5m"
-    } else {
-      "lt5m"
+    hashMapOf(
+      "task.invocations.duration" to taskInvocationsId.tags(),
+      "task.invocations.duration.withType" to taskInvocationsWithTypeId.tags()
+    ).forEach {
+      name, tags ->
+        val id = registry.createId(name).withTags(tags)
+        registry.timer(id).record(elapsedMillis, TimeUnit.MILLISECONDS)
```

Comment:
Is this valid use of a timer given we'll repeatedly generate a value for the same id every time this handler is invoked?

Reply:
It will just update the stats for the timer with that id, so the usage of the timer should be fine. However, based on the duration strings, I wonder if a long task timer would be more appropriate for this use-case, see:

http://netflix.github.io/spectator/en/latest/intro/timer/

Reply:
Thanks for the suggestion @brharrington, looks like this is not going to work in this context because tasks can get started and completed on different instances